src/bot.ts

```diff
@@ -389,6 +392,7 @@ export class DiscordBot {
             }
             const channel = guild.channels.get(room);
             if (channel && channel.type === "text") {
+                this.ClientFactory.bindMetricsToChannel(channel as Discord.TextChannel);
```

Comment:
shouldn't we  always bind the channel on message processing? Or even better, right when the message enters via the event listeners?

Reply:
Nope, because we only call the *send* function when sending stuff that came via matrix. Though I guess we should also bind in the event listeners too.